### PR TITLE
`azurerm_api_management_api_policy`, `azurerm_api_management_api_operation_policy`, `azurerm_api_management_product_policy` : fix parsing resource id error

### PR DESCRIPTION
**Pull request by sinbai**
For resources `azurerm_api_management_api_policy`, `azurerm_api_management_api_operation_policy`, `azurerm_api_management_product_policy`, after importing the resource using a version before v3.70.0 of Terraform Provider,  the id in state file ends with "/policies/policy", and the id in state file ends with "/policies/xml" when creating resource by Terraform. Therefore, when migrating Pandora SDK, both cases need to be migrated.

 In [PR #22783](https://github.com/hashicorp/terraform-provider-azurerm/pull/22783) only the case where the ID ends with "/policies/xml" is processed, so submit this PR to process the case where the ID ends with "/policies/policy" to solve the parsing id error.

Might fix issue #23112, #23095.

Local manual testing, the steps are as follows:
	

> 1. Import  the existing resources `azurerm_api_management_api_policy`, `azurerm_api_management_api_operation_policy`, `azurerm_api_management_product_policy` using TF provider v3.69.0. 
> 	2. Upgrade TF provider to v3.71.0.
> 	3. Run "terraform plan" => The parsing id error occurred.
> 	4. Upgrade the TF provider to the current PR build version.
> 	5.  Run "terraform plan"  => No error.

### CODE REVIEW

File: internal/services/apimanagement/migration/api_product_policy_v1_to_v2.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		// new id : /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.ApiManagement/service/service1/products/exampleId
+		oldId := rawState["id"].(string)
+
+		// Prior to v3.70.0 of Terraform Provider, after importing resource, the id in state file ends with "/policies/policy", the id in state file ends with "/policies/xml" for creating resource by Terraform.
+		// So after migrating pandora SDK (starting from v3.70.0), these two cases need to be migrated.
+		// In ApiManagementApiPolicyV0ToV1, only the case where the ID ends with "/policies/xml" is processed, so the case where the ID ends with "/policies/policy" is processed here to solve the parse id error.
+		newId := strings.TrimSuffix(oldId, "/policies/policy")
+
+		log.Printf("[DEBUG] Updating ID from %q tmakeo %q", oldId, newId)
+		rawState["id"] = newId
```
Comment (tombuildsstuff) [inline internal/services/apimanagement/migration/api_product_policy_v1_to_v2.go:60]: (same here / as above)

File: internal/services/apimanagement/migration/api_operation_policy_v1_to_v2.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		// new id : /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.ApiManagement/service/instance1/apis/api1/operations/operation1
+		oldId := rawState["id"].(string)
+
+		// Prior to v3.70.0 of Terraform Provider, after importing resource, the id in state file ends with "/policies/policy", the id in state file ends with "/policies/xml" for creating resource by Terraform.
+		// So after migrating pandora SDK (starting from v3.70.0), these two cases need to be migrated.
+		// In ApiManagementApiPolicyV0ToV1, only the case where the ID ends with "/policies/xml" is processed, so the case where the ID ends with "/policies/policy" is processed here to solve the parse id error.
+		newId := strings.TrimSuffix(oldId, "/policies/policy")
+
+		log.Printf("[DEBUG] Updating ID from %q tmakeo %q", oldId, newId)
+		rawState["id"] = newId
```
Comment (tombuildsstuff) [inline internal/services/apimanagement/migration/api_operation_policy_v1_to_v2.go:67]: (same here)

```suggestion
		newId := strings.TrimSuffix(oldId, "/policies/policy")

		log.Printf("[DEBUG] Updating ID from %q to %q", oldId, newId)
		rawState["id"] = newId
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @sinbai 

Thanks for this PR - taking a look through on the whole this looks pretty good but since we're updating these IDs we should also look to parse/reformat the ID too to ensure that the `resourceGroups` and other segments are correct at the same time. However if we can fix that up then this should otherwise be good to go 👍 

Thanks!

File: internal/services/apimanagement/migration/api_product_policy_v1_to_v2.go (hunk 1 above)
Comment (sinbai) [reply internal/services/apimanagement/migration/api_product_policy_v1_to_v2.go]: Fixed.

File: internal/services/apimanagement/migration/api_operation_policy_v1_to_v2.go (hunk 2 above)
Comment (sinbai) [reply internal/services/apimanagement/migration/api_operation_policy_v1_to_v2.go]: Fixed.

File: internal/services/apimanagement/migration/api_policy_v1_to_v2.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		// new id : /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.ApiManagement/service/service1/apis/exampleId
+		oldId := rawState["id"].(string)
+
+		// Prior to v3.70.0 of Terraform Provider, after importing resource, the id in state file ends with "/policies/policy", the id in state file ends with "/policies/xml" for creating resource by Terraform.
+		// So after migrating pandora SDK (starting from v3.70.0), these two cases need to be migrated.
+		// In ApiManagementApiPolicyV0ToV1, only the case where the ID ends with "/policies/xml" is processed, so the case where the ID ends with "/policies/policy" is processed here to solve the parse id error.
+		newId := strings.TrimSuffix(oldId, "/policies/policy")
+
+		log.Printf("[DEBUG] Updating ID from %q to %q", oldId, newId)
+		rawState["id"] = newId
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/apimanagement/migration/api_policy_v1_to_v2.go]: Fixed.

**Comment by sinbai** [comment]
> hey @sinbai
> 
> Thanks for this PR - taking a look through on the whole this looks pretty good but since we're updating these IDs we should also look to parse/reformat the ID too to ensure that the `resourceGroups` and other segments are correct at the same time. However if we can fix that up then this should otherwise be good to go 👍
> 
> Thanks!

Hi @tombuildsstuff , thanks for your feedback. The code has been updated. Could you please take another look?

**Pull request merged**